fdbclient/SpecialKeySpace.actor.cpp

```diff
@@ -1803,9 +1793,20 @@ ACTOR static Future<RangeResult> CoordinatorsAutoImplActor(ReadYourWritesTransac
 		throw special_keys_api_failure();
 	}
 
-	for (const auto& address : _desiredCoordinators) {
-		autoCoordinatorsKey += autoCoordinatorsKey.size() ? "," : "";
-		autoCoordinatorsKey += address.toString();
+	if (result == CoordinatorsResult::SAME_NETWORK_ADDRESSES) {
+		for (const auto& host : old.hostnames) {
+			autoCoordinatorsKey += autoCoordinatorsKey.size() ? "," : "";
+			autoCoordinatorsKey += host.toString();
+		}
+		for (const auto& coord : old.coords) {
+			autoCoordinatorsKey += autoCoordinatorsKey.size() ? "," : "";
+			autoCoordinatorsKey += coord.toString();
+		}
+	} else {
+		for (const auto& address : _desiredCoordinators) {
```

Comment:
IIUC, if the old connection string is not valid, we will always switch to using IP as connection string, correct?

Also, it seems that `auto coordinators` command will always use IP as connection string.

Reply:
If the old connection string is not valid, or cannot satisfy requirements (e.g., the redundency requirement), we will switch to use IP.

Yes, `auto coordinators` command will always use IP as connection string, because the coordinators are selected from workers' IP addresses.